### PR TITLE
Melhora forma como erros são lançados

### PR DESCRIPTION
**Pull request by givigier**
Melhorias:

* Lança a exceção CEPNotFound quando o cep não foi encontrado ou não foi informado.
* Lança a exceção InvalidCEPFormat quando o formato do cep é inválido
* Adiciona versão das dependências para prevenir warnings ao fazer o build da gem

### CODE REVIEW

**Comment by pdf13** [comment]
+1

**Comment by prodis** [comment]
Related to #10 issue.

File: lib/correios/cep/address_finder.rb [hunk 1]
```diff
@@ -19,6 +22,21 @@ def web_service
       def parser
         @parser ||= Correios::CEP::Parser.new
       end
+
+      private
+
+      def format_response(response)
```
Comment (prodis) [inline lib/correios/cep/address_finder.rb]: This method do not really format the response, after its execution response remains the same. A better name is desirable.

File: lib/correios/cep/parser.rb [hunk 2]
```diff
@@ -15,6 +15,10 @@ class Parser
 
       def address(xml)
         doc = Ox.parse(xml)
+        
+        errors = get_errors(doc)
+        return errors unless errors.nil?
```
Comment (prodis) [inline lib/correios/cep/parser.rb]: `errors` variable is a `String`. It is expected `Parser#address` method returns a `Hash`.

File: lib/correios/cep/address_finder.rb [hunk 3]
```diff
@@ -19,6 +22,21 @@ def web_service
...
+        case response
+        when "CEP NAO ENCONTRADO" 
+          raise CEPNotFound.new "" 
+        when "BUSCA DEFINIDA COMO EXATA, 0 CEP DEVE TER 8 DIGITOS" 
+          raise InvalidCEPFormat.new "" 
+        when "CEP NAO INFORMADO"
+          raise CEPNotFound.new ""
```
Comment (prodis) [inline lib/correios/cep/address_finder.rb]: Required zip code differs of zip code not found, it should not treated in the same way.

**Comment by prodis** [comment]
I don't think the best way to deal with a zip code not found is raising an exception.

Exceptions should be used for unexpected situations. In this case, it is expected that an user supplies a non existing zip code in some situation. We can deal a zip code not found returning an empty result. 

From the perspective of the code that uses the gem, the application client need to handle `CEPNotFound` exception in each `Correios::CEP::AddressFinder.get` call. That is not so good.

An example below:

```ruby
require 'correios-cep'

begin
  address = Correios::CEP::AddressFinder.get('99999999')
  # Logic when zip code found. 
rescue Correios::CEP::AddressFinder::CEPNotFound => e
  # Logic when zip code not found.
end
```

About `InvalidCEPFormat` exception, we can validate zip code input without to do a HTTP request. A regular expression is enough to check zip code format. `ArgumentError` exception is indicated to use in input validation.

Anyway I revised and comment the pull request.

Thank you for your submission.

File: lib/correios/cep/address_finder.rb (hunk 1 above)
Comment (givigier) [inline lib/correios/cep/address_finder.rb]: @prodis I agree with you. I'll change the name of this method and send another PR.

File: lib/correios/cep/parser.rb (hunk 2 above)
Comment (givigier) [inline lib/correios/cep/parser.rb]: Sure. I'll fix this too.

File: lib/correios/cep/address_finder.rb (hunk 3 above)
Comment (givigier) [inline lib/correios/cep/address_finder.rb]: What about create CEPEmpty and raise this exception when cep is empty?

**Comment by givigier** [comment]
@prodis I understand your point. I still guess that exceptions are a good way to inform that a CEP is invalid or was not found. For example, if I have to import hundreds CEPs from a spreadsheet to my database. If I got only an empty hash I don't know if correios-cep returned an empty hash because the format of CEP is invalid, because the webservice didn't respond, because CEP was not found or because I tried to find an address passing an empty CEP. When I use exceptions I can catch a specific error on each case and show a better feedback to my user.

**Pull request closed without merging**